core/src/main/scala/org/ensime/indexer/SearchService.scala

```diff
@@ -34,6 +34,11 @@ class SearchService(
     with FileChangeListener
     with SLF4JLogging {
 
+  private[indexer] def isUserFile(file: FileName): Boolean = {
+    //may want to cache the mapped results. 
+    (config.allTargets map { vfs.vfile }) exists { file isAncestor _.getName }
```

Comment:
no need to cache, that should be handled at the layer beneath us already (of course, it isn't, but that's because VFS is crap...)